### ISSUE TITLE
Error "contextList is empty" after signing in VSO

### ISSUE THREAD

**Issue by ias0nas** [Microsoft/vso-intellij#104]
While on the Checkout from Team Services window and after signing successfully I get an error message above the buttons reading "contextList is empty".

Same credentials in Visual Studio bring up the project I have access to on visualstudio.com

Any logs I can check for more details?

**Comment by leantk** [comment]
Hi @ias0nas,

You can turn on debugging and get a log by following the last FAQ here: https://java.visualstudio.com/Docs/tools/intelliJ#_frequently-asked-questions-faq

It looks like something must have gone wrong during signing in because that should create a context that populates the contextList but without more information I can't tell you exactly what went wrong. If you attach the log here or email it to me at leantk@microsoft.com I can help you debug what might have happened.

Thanks,
-Leah

**Comment by ias0nas** [comment]
Hi @leantk 

Not sure these instructions apply for Rider?
Doesn't seem to be options for custom log configuration.

I am using Rider (on Linux if that makes any difference).
Does it have another way of enabling logging?

Thanks

![contextlistempty](https://user-images.githubusercontent.com/9936844/29994866-67ba4d22-8fd2-11e7-9dbe-730316ee406a.gif)

**Comment by leantk** [comment]
Hi @ias0nas,

Sorry for the confusion with the logging. The toolbar for Rider only shows up when you have a project opened already. If you have another project that you could open, you could see the toolbar across the top that has the Help dropdown. Once the debugging is enabled, could you send me the entire log please? 

Another option you can try is going to the Team Foundation Server tab and pasting your Team Services repo URL in the Server URL text box. I know it looks like it's just for servers but it also will take your repo URL that you want to clone. Let me know if this gives you a different outcome then what you saw with the Team Services tab.

Thanks,
-Leah

**Comment by ias0nas** [comment]
Hi @leantk 

Unfortunately the same error is returned when trying to sign in to TFS after having opened a project.
I enabled the debugging as described and this is the output:

```
20:21:15.754 | INFO  | SolutionLifecycleHost          | Full warm solution load with caches took 47309 milliseconds.
20:25:12.443 | INFO  | VsoCheckoutPageModel           | Skipping loading contexts in constructor
20:25:13.660 | INFO  | ServerPollingManager           | Timer fired
20:25:13.661 | INFO  | ServerEventManager             | triggering ALL events
20:25:13.661 | INFO  | ServerEventManager             | triggering event: BUILDS_CHANGED
20:25:13.661 | INFO  | ServerEventManager             | triggering event: PULL_REQUESTS_CHANGED
20:25:13.661 | INFO  | ServerEventManager             | triggering event: WORK_ITEMS_CHANGED
20:25:20.240 | INFO  | VsoCheckoutPageModel           | loadRepositories called
20:25:20.258 | INFO  | VsoAuthenticationProvider      | authenticateAsync for server: https://app.vssps.visualstudio.com
20:25:20.439 | INFO  | VstsPatAuthenticator           | Signing out from uri: https://app.vssps.visualstudio.com
20:25:20.450 | INFO  | OAuth2Authenticator            | Ready to launch browser flow to retrieve oauth2 token.
20:25:20.452 | INFO  | SettingsHelper                 | Searching for /home/ias0nas/.microsoft/VstsAuthLib4J/settings.properties
20:25:20.454 | INFO  | OAuth2Authenticator            | Attempt to use oauth2-useragent provider: JavaFx
20:25:20.462 | INFO  | OAuth2Authenticator            | Using oauth2-useragent providers to retrieve AAD token.
20:25:54.278 | INFO  | HttpProxyServiceImpl           | useHttpProxy: false
20:25:56.143 | INFO  | VsoAuthenticationProvider      | getAuthenticationInfoAsync succeeded
20:25:56.154 | INFO  | HttpProxyServiceImpl           | useHttpProxy: false
20:25:57.525 | INFO  | ServerContextLookupListener    | loadContexts activeOperation with 0 contexts and a scope of PROJECT
20:25:57.531 | INFO  | VsoCheckoutPageModel           | signOut called
20:25:57.653 | INFO  | VstsPatAuthenticator           | Signing out from uri: https://app.vssps.visualstudio.com
```

**Comment by ias0nas** [comment]
@leantk 
Using the Team Foundation Server as you suggested tab works fine for the same credentials so probably something going on with the first tab.

If you need any more logs or anything else, let me know.

Thanks

**Comment by leantk** [comment]
@ias0nas I'm glad to hear things worked with the second tab. In some scenarios, a user's username/pw can't find their associated VSTS accounts which is what the first tab (VSTS) tries. The second tab (TFS) is able to use the repo URL provided to make the association. It's something that we are aware of and are looking into but for now the TFS tab is way to unblock users. 

I'm going to close this issue since we already are tracking that larger problem and since you're unblocked. If you run into another issue please feel free to reopen this.

Thanks,
-Leah

**Comment by nfantone** [comment]
@leantk Same thing happened to me today. Unfortunately, workaround was a no-go for me. I'm not exactly sure I have a repo URL to use with the "Team Foundation Server" tab, though.

Any new ideas?

**Comment by leantk** [comment]
@nfantone are you using Azure DevOps (cloud offering) or do you have TFS (on-premises offering)? If it's Azure DevOps your repo URL would look something like https://dev.azure.com/<Org_Name>/_git/<Repo_Name> or https://<Org_Name>.visualstudio.com/_git/<Repo_Name>. Basically the URL you see when you browse to the repo view. Did you try putting that in the "Team Foundation Server" tab's server URL box and logging in that way?

**Comment by ForNeVeR** [comment]
@aleybe, first of all, could you please try a suggested workaround? Try to authenticate using the Team Foundation Server tab. If it doesn't work, could you please provide us with more details?

**Comment by aleybe** [comment]
Hi Friedrich,

The work around works yes, but this means I have to copy paste the
visualstudio.com address into the window every time.
It would be awesome if this worked more like visual studio where connecting
to the DevOps instance allowed us to pick from our repos.

Thanks,
Alex

**Comment by ForNeVeR** [comment]
@aleybe, thanks, I see. Any ideas why it couldn't work with your account from the first tab? Anything unusual with your authentication or Azure DevOps organization / account setup?

@leantk, sorry to bother you, but are there any public details on this authentication issue? Is it something we could try to fix here or implement a workaround?

**Comment by aleybe** [comment]
Hi Freidrich,

I suppose we do have a bit of a different authentication.

We login to domainA.visualstudio.com with aleybe@domainB.com ... The two
domains are federated.
I believe this is what is causing the issue as if I put aleybe@domainB.com
Rider is assuming I'm trying to login to domainB.visualstudio.com instead
did domainA visualstudio.com

If that makes sense :p

**Comment by ForNeVeR** [comment]
@aleybe, thanks for information. I've extracted that information to the new issue: #282.
